s3pypi/storage.py

```diff
@@ -1,62 +1,58 @@
-import logging
-import os
+from pathlib import Path
+from typing import Optional
 
 import boto3
-from botocore.exceptions import ClientError
+import botocore
 
-from s3pypi.package import Index
+from s3pypi.index import Index
 
-log = logging.getLogger()
-
-
-class S3Storage(object):
-    """Abstraction for storing package archives and index files in an S3 bucket."""
 
+class S3Storage:
     def __init__(
         self,
-        bucket,
-        secret=None,
-        region=None,
-        bare=False,
-        acl="public-read",
-        profile=None,
+        bucket: str,
+        profile: Optional[str] = None,
+        region: Optional[str] = None,
+        prefix: Optional[str] = None,
+        acl: Optional[str] = None,
+        s3_put_args: Optional[dict] = None,
+        unsafe_s3_website: bool = False,
     ):
-        if profile:
-            boto3.setup_default_session(profile_name=profile)
-        self.s3 = boto3.resource("s3", region_name=region)
+        session = boto3.Session(profile_name=profile, region_name=region)
+        self.s3 = session.resource("s3")
         self.bucket = bucket
-        self.secret = secret
-        self.index = "" if bare else "index.html"
-        self.acl = acl
-
-    def _object(self, package, filename):
-        path = "%s/%s" % (package.directory, filename)
-        return self.s3.Object(
-            self.bucket, "%s/%s" % (self.secret, path) if self.secret else path
+        self.prefix = prefix
+        self.index_name = "index.html" if unsafe_s3_website else ""
```

Comment:
Can it be changed to detect the presence of `index.html` file and use it if already exists? With current implementation it will break when there are 2 different packages being uploaded to one s3pypi repository, using 2 different s3pypi package versions. It is not always possible to sync every single package to use the same s3pypi version, so this kind of backwards compatibility IMO should be handled with care.

Reply:
Alternatively, a new release can be provided for the old branch, without introducing any other changes but the detection of `bare` index file and using it even if `--bare` is not provided. This change can be easier synchronized between projects.